File: bitget-node-sdk-api/src/lib/v1/MixOrderApi.ts
import {BaseApi} from '../BaseApi';

export class MixOrderApi extends BaseApi {

    placeOrder(qsOrBody: object) {
        const url = '/api/mix/v1/order/placeOrder';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    batchPlaceOrder(qsOrBody: object) {
        const url = '/api/mix/v1/order/batch-orders';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    cancelOrder(qsOrBody: object) {
        const url = '/api/mix/v1/order/cancel-order';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    batchCancelOrders(qsOrBody: object) {
        const url = '/api/mix/v1/order/cancel-batch-orders';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    ordersPending(qsOrBody: object) {
        const url = '/api/mix/v1/order/current';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    ordersHistory(qsOrBody: object) {
        const url = '/api/mix/v1/order/history';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    fills(qsOrBody: object) {
        const url = '/api/mix/v1/order/fills';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    placePlanOrder(qsOrBody: object) {
        const url = '/api/mix/v1/plan/placePlan';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    cancelPlanOrder(qsOrBody: object) {
        const url = '/api/mix/v1/plan/cancelPlan';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    ordersPlanPending(qsOrBody: object) {
        const url = '/api/mix/v1/plan/currentPlan';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    ordersPlanHistory(qsOrBody: object) {
        const url = '/api/mix/v1/plan/historyPlan';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    traderCloseOrder(qsOrBody: object) {
        const url = '/api/mix/v1/trace/closeTrackOrder';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    traderCurrentOrders(qsOrBody: object) {
        const url = '/api/mix/v1/trace/currentTrack';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    traderHistoryTrack(qsOrBody: object) {
        const url = '/api/mix/v1/trace/historyTrack';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    followerCloseByTrackingNo(qsOrBody: object) {
        const url = '/api/mix/v1/trace/followerCloseByTrackingNo';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    followerQueryCurrentOrders(qsOrBody: object) {
        const url = '/api/mix/v1/trace/followerOrder';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    followerQueryHistoryOrders(qsOrBody: object) {
        const url = '/api/mix/v1/trace/followerHistoryOrders';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }
}

File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/v1/MixOrderApi.java
package com.bitget.openapi.api.v1;

import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.*;

import java.util.Map;

public interface MixOrderApi {

    // normal order
    @POST("/api/mix/v1/order/placeOrder")
    Call<ResponseResult> placeOrder(@Body Map<String, String> paramMap);

    @POST("/api/mix/v1/order/batch-orders")
    Call<ResponseResult> batchOrders(@Body Map<String, String> paramMap);

    @POST("/api/mix/v1/order/cancel-order")
    Call<ResponseResult> cancelOrder(@Body Map<String, String> paramMap);

    @POST("/api/mix/v1/order/cancel-batch-orders")
    Call<ResponseResult> cancelBatchOrder(@Body Map<String, String> paramMap);

    @GET("/api/mix/v1/order/history")
    Call<ResponseResult> history(@QueryMap Map<String, String> paramMap);

    @GET("/api/mix/v1/order/current")
    Call<ResponseResult> current(@QueryMap Map<String, String> paramMap);

    @GET("/api/mix/v1/order/fills")
    Call<ResponseResult> fills(@QueryMap Map<String, String> paramMap);


    // plan
    @POST("/api/mix/v1/plan/placePlan")
    Call<ResponseResult> placePlan(@Body Map<String, String> paramMap);

    @POST("/api/mix/v1/plan/cancelPlan")
    Call<ResponseResult> cancelPlan(@Body Map<String, String> paramMap);

    @GET("/api/mix/v1/plan/currentPlan")
    Call<ResponseResult> currentPlan(@QueryMap Map<String, String> paramMap);

    @GET("/api/mix/v1/plan/historyPlan")
    Call<ResponseResult> historyPlan(@QueryMap Map<String, String> paramMap);


    // trader
    @POST("/api/mix/v1/trace/closeTrackOrder")
    Call<ResponseResult> traderCloseOrder(@Body Map<String, String> paramMap);

    @GET("/api/mix/v1/trace/currentTrack")
    Call<ResponseResult> traderCurrentOrders(@QueryMap Map<String, String> paramMap);

    @GET("/api/mix/v1/trace/historyTrack")
    Call<ResponseResult> traderHistoryTrack(@QueryMap Map<String, String> paramMap);

    @POST("/api/mix/v1/trace/followerCloseByTrackingNo")
    Call<ResponseResult> followerCloseByTrackingNo(@Body Map<String, String> paramMap);

    @GET("/api/mix/v1/trace/followerOrder")
    Call<ResponseResult> followerOrder(@QueryMap Map<String, String> paramMap);

    @GET("/api/mix/v1/trace/followerHistoryOrders")
    Call<ResponseResult> followerHistoryOrders(@QueryMap Map<String, String> paramMap);
}
